readthedocs/organizations/views/public.py

```diff
@@ -36,28 +42,57 @@ class DetailOrganization(OrganizationView, DetailView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         org = self.get_object()
-        context['projects'] = (
+        projects = (
             Project.objects
             .for_user(self.request.user)
             .filter(organizations=org)
             .all()
         )
-        context['teams'] = (
+        teams = (
             Team.objects
             .member(self.request.user, organization=org)
             .prefetch_related('organization')
             .all()
         )
-        context['owners'] = org.owners.all()
+        if settings.RTD_EXT_THEME_ENABLED:
+            filter = OrganizationProjectListFilterSet(
+                self.request.GET,
+                request=self.request,
+                queryset=projects,
+                teams_queryset=teams,
```

Comment:
I'm guessing you are passing both querysets, so they can be re-used in the context of the old templates. I like more the idea of passing the object that the querysets need to be filtered by, like in

https://github.com/readthedocs/readthedocs.org/blob/e795e57c9e637aebe6fc55d36777fcf3a61e28e1/readthedocs/organizations/filters.py#L275

The filterset class could expose those querysets like `OrganizationProjectListFilterSet.get_teams`, so it can be re-used in the context. But not sure if that will execute the query twice (maybe cache the method if that happens).

Just an idea, not sure if this will end up being more tedious than just passing the queryset.

Reply:
I also leaned this way and would consider passing in `organization` instead of the extra queryset instance(s) to be a bit cleaner.

I started this direction too, but did realize that this would be confusing as all of these querysets are already defined and evaluated at the view level. There is potential to duplicate queries though, yeah.

Duplicate queries might not even be a concern though. The filters aren't used on the current dashboard and I don't think all of the view queries will be used by the views once the new dashboard is the only dashboard.

I'm solidly impartial at this point I think. If folks feel it's a better design to define these querysets at the `FilterSet` level, I'm :+1: